src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const state2 = { ...state };
+  const log = [];
+
+  for (const action in actions) {
+    switch (actions[action].type) {
+      case 'addProperties':
+        Object.assign(state2, actions[action].extraData);
+
+        const log2 = { ...state2 };
```

Comment:
taking into account that you're anyway doing copy using `spread` operator, you can add here extra data as well, and don't do object.assign in the 16 row
in this case, you won't mutate the state object, so the 10 row will be unnecessarily too

Reply:
fixed mistakes but can`t avoid copying the state at the beginning it can lead to the mistakes in other scenarios